### PR TITLE
chore(updatecli) update cst file synchronously with nginx update

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/docker-confluence-data/pull/64#issuecomment-2110447675
the cst file need to be updated too

### CODE REVIEW

**Comment by smerle33** [comment]
```
updateTestHarness
-----------------

**Dry Run enabled**

"cst.yml" updated with [dry run] content "expectedError: [\"1.26.0\"]\n"

```
--- cst.yml
+++ cst.yml
@@ -4,4 +4,4 @@
   - name: "Check Nginx Version"
     command: "nginx"
     args: ["-v"]
-    expectedError: ["1.24."]
+    expectedError: ["1.26.0"]

```

⚠ - 1 file(s) updated with "expectedError: [\"1.26.0\"]\n":
        * cst.yml
 ```

File: updatecli/updatecli.d/nginx.yaml [hunk 1]
```diff
@@ -30,6 +30,17 @@ sources:
...
+    name: Get Main Version latest stable version of nginx
+    kind: gittag
+    scmid: nginxGithubMirror
+    spec:
+      versionfilter:
+        kind: regex
+        ## Nginx stable version have the minor digit as an even number
+        pattern: 'release-(\d+)\.(\d*[0|2|4|6|8])\.(\d+)'
+    transformers:
+      - trimprefix: "release-"
```
Comment (dduportal) [inline updatecli/updatecli.d/nginx.yaml:43]: Is there a reason to create a second source of (almost) the same value?

I would have removed the `- addsuffix: "-alpine"` transformer in the existing source, and would have added it (the `addsuffix` transformer) to the condition and the target where it is required instead.

Comment (smerle33) [reply updatecli/updatecli.d/nginx.yaml]: you're completely right

**Review by dduportal — CHANGES_REQUESTED** [review]
You forgot to update the existing target:

```
⚠ The line #1, matched by the keyword "FROM" and the matcher "nginx", is changed from "FROM nginx:1.26.0-alpine" to "FROM nginx:1.26.0".
```

=> this change is unwanted

File: updatecli/updatecli.d/nginx.yaml [hunk 2]
```diff
@@ -52,7 +52,17 @@ targets:
...
+    name: "Update the expectedError version in the test harness"
+    sourceid: latestStableRelease
+    scmid: default
+    kind: file
+    spec:
+      file: "cst.yml"
+      matchpattern: >
+        expectedError: \["(.*)"]
+      replacepattern: >
+        expectedError: ["{{ source "latestStableRelease" }}"]
```
Comment (dduportal) [inline updatecli/updatecli.d/nginx.yaml:65]: Is there a reason for not using the Yaml parser?

Comment (smerle33) [reply updatecli/updatecli.d/nginx.yaml:65]: sorry, I forgot to switch to draft, I am still working on the major.minor version without the patch and it imply using the kind yaml

**Comment by smerle33** [comment]
sounds good now : 
```
TARGETS
========

updateReleaseInDockerfile
-------------------------

**Dry Run enabled**

✔ The line #1, matched by the keyword "FROM" and the matcher "nginx", is correctly set to "FROM nginx:1.26.0-alpine".
✔ - changed lines [] of file "/var/folders/_k/4csqmqxd4qq4243_4dsbdp5m0000gn/T/updatecli/github/jenkins-infra/docker-confluence-data/Dockerfile"

updateTestHarness
-----------------

**Dry Run enabled**

⚠ - change detected:
        * key "$.commandTests[0].expectedError[0]" should be updated from "\"1.24.\"" to "\"1.26.\"", in file "cst.yml"
```

**Pull request merged**